taxonomy-editor-frontend/src/pages/export/index.jsx

```diff
@@ -0,0 +1,145 @@
+            // Download taxonomy
+            var url = window.URL.createObjectURL(blob);
+            var a = document.createElement('a');
+            a.href = url;
+            a.download = taxonomyName + '.txt'
+            document.body.appendChild(a);
+            a.click(); a.remove();
+
+            setLoadingForDownload(false);
+        }).catch(() => {
+            setErrorMessage("Download failed!")
+            setLoadingForDownload(false); setOpenSuccessDialog(true)
+        })
+    }
+
+    function handleGithub() {
+        setLoadingForGitHub(true);
+        fetch(baseURL+'githubexport', {
+            method: 'GET'
+        }).then(async (response) => {
+            const responseBody = await response.json();
+            if (!response.ok) {
+                throw Error(responseBody.detail);
+            } else {
+                setPullRequestURL(responseBody);
+                setLoadingForGitHub(false);
+                setOpenSuccessDialog(true);
+            }
+        }).catch((detail) => {
+            setErrorMessage(detail.message); 
+            setLoadingForGitHub(false); setOpenErrorSnackbar(true)
+        })
+    }
+
+    function LoadingButton(props) {
+        const { onClick, loading, text, sx } = props;
+        return (
+            <Button variant="contained" sx={sx} onClick={onClick} disabled={loading}>
+                {loading && <CircularProgress size={24} />}
+                {!loading && text}
+            </Button>
+        );    
+    }
+
+    function handleCloseErrorSnackbar() {
+        setOpenErrorSnackbar(false);
+    }
+    function handleCloseSuccessDialog() {
+        setOpenSuccessDialog(false);
+    }
+
+    return (
+        <Box>
+            <Grid
+            container
+            direction="column"
+            alignItems="center"
+            justifyContent="center"
+            >
+                <Typography sx={{mt: 4}} variant="h3">Export Taxonomy</Typography>
+                <Typography sx={{mt: 8}} variant="h5">Click the button below to download your edited taxonomy</Typography>
+                <LoadingButton
+                    loading={loadingForDownload}
+                    onClick={handleDownload}
+                    sx={{mt: 4, width: 130}}
+                    text="Download"
+                >
+                </LoadingButton>
+                <Typography sx={{mt: 10}} variant="h5">Click the button below to create a Pull Request to Github</Typography>
+                <LoadingButton
+                    loading={loadingForGithub}
+                    onClick={handleGithub}
+                    sx={{mt: 4, width: 130}}
+                    text="Create PR"
+                >
+                </LoadingButton>
+                
+            </Grid>
+            {/* Snackbar to show errors */}
+            <Snackbar anchorOrigin={{vertical: 'top', horizontal: 'right'}} open={openErrorSnackbar} autoHideDuration={3000} onClose={handleCloseErrorSnackbar}>
```

Comment:
what about using the errorMessage for open and close? That way we will have one state that controls both the open/close and the message being shown, so instead of setting two values we'd set only 1.

Reply:
Yep, removed the duplicate state.